modules/fedimint-wallet/src/lib.rs

```diff
@@ -339,12 +343,14 @@ impl ServerModulePlugin for Wallet {
     }
 
     async fn consensus_proposal<'a>(&'a self) -> Vec<Self::ConsensusItem> {
+        let dbtx = self.read_only_db.begin_transaction();
```

Comment:
Why is this using the read_only_db? Is this consensus critical?

Reply:
It kinda is, because we ban peers for not contributing expected signatures etc. But since it's called after all other DB tx from the previous epoch have been committed and no consensus relevant changes to the DB happen in parallel I think it's ok.

My general understanding is: if it's consensus critical and there is a DB tx with uncommitted changes we need to read from the DB tx (particularly when applying inputs/outputs) to prevent e.g. double spends. If there's no DB tx in flight that contains consensus critical changes we can open a new one (like for `consensus_proposal`).

For non-consensus critical stuff like API requests consistency would be nice, but isn't a problem in practice imo and can wait.